ncm-metaconfig/src/main/metaconfig/udev/pan/rule_schema/schema.pan

```diff
@@ -0,0 +1,105 @@
+declaration template metaconfig/udev/rule_schema/schema;
+
+include 'pan/types';
+
+@documentation{
+Schema for udev rules which consist of match or matches of key that results in
+assignment or actions of matched key(s), referred as match_rule and udev_assign_rule in this schema.
+
+match key is either dict of dict with a key and without one that has a value and an operator with 
+equality check.
+matched key result in action/assigment in ether dict of dict with a key or without one that has a
+ value and an operator with assignment.
+
+There could be multiple lines of rules with different matches independent of each rules.
+
+example:
+/etc/udev/rules.d/00-x710.rules
+
+ACTION=="add|change", ATTRS{device}=="0x1572", RUN+="/sbin/ethtool -s '%k' speed 10000"
```

Comment:
Hmm, the `}` here terminates the annotation, everything after it is treated as pan code, which then fails with a parse error and there doesn't appear to be any way to escape this.

Perhaps as @ned21 suggested this is actually too complex for an annotation.

Reply:
```suggestion
ACTION=="add|change", ATTRS{device}=="0x1572", RUN+="/sbin/ethtool -s '%k' speed 10000"

'rule' = dict(
    'match_rule', dict(
        'attrs', list(
            dict(
                'key', 'device',
                'operator', '==', 
                'value', '0x1572',
            ),
        ),
    ),
    'udev_assign_rule': dict(
        'run', list(
             dict(
                'operator', '+=', 
                'value', "/sbin/ethtool -s '%k' speed 10000",
            ),
        ),
    ),
);
```